extensions/amp-youtube/0.1/amp-youtube.js

```diff
@@ -29,13 +29,34 @@ const YT_PLAYER_STATE_PLAYING = 1;
 
 class AmpYoutube extends AMP.BaseElement {
 
-  /** @override */
-  preconnectCallback(onLayout) {
-    this.preconnect.url('https://www.youtube.com', onLayout);
+  constructor() {
```

Comment:
Good catch. Done.

Also: Yes, will need to declare all instance vars in constructor. Better for type checking and performance.